js/shaders/scatter-fragment.glsl

```diff
@@ -35,10 +35,13 @@ vec2 rotate_xy(vec2 x, float angle) {
 
 /*
  * Returns 1.0 if pixel inside of a circle (0.0 otherwise) given the circle radius and the
- * pixel distance from the circle center.
+ * pixel position.
  */
-float circle(in float radius, in float dist) {
-    return 1.0 - smoothstep(radius - SMOOTH_PIXELS, radius + SMOOTH_PIXELS, dist);
+float circle(in float radius, in vec2 pixel_position) {
+    float d = pow(pixel_position.x, 2.0) + pow(pixel_position.y, 2.0);
```

Comment:
isn't this equivalent to `dot(pixel_position, pixel_position)`, which might be faster it the optimization path doesn't recognize it.

Reply:
It is! Good catch